File: development/NET/samples/client/DABrowse/VB/MyDaSession.vb
Imports Softing.OPCToolbox.Client

Namespace DABrowse
	Public Class MyDaSession
		Inherits DaSession

#Region "Constructor"
		'-----------------

		Public Sub New(ByVal url As String)
			MyBase.New(url)
			AddHandler StateChangeCompleted, AddressOf HandleStateChanged
		End Sub

		'--
#End Region

#Region "Private Members"
		'---------------------
		Dim m_connected As Boolean
#Region "Private Attributes"
		'------------------------


		'--
#End Region

		'--
#End Region

#Region "Public Methods"
		'---------------------

		'--
#End Region

#Region "Public Properties"
		'-----------------------
		Public Function isConnected() As Boolean
			Return m_connected
		End Function

		'--
#End Region

#Region "Handles"
		'---------------------
		Public Sub HandleStateChanged(ByVal sender As ObjectSpaceElement, ByVal state As EnumObjectState)

			If state = EnumObjectState.CONNECTED Then
				m_connected = True
			Else
				m_connected = False
			End If
		End Sub


		'--
#End Region

	End Class

End Namespace

File: development/NET/samples/client/DABrowse/CS/MyDaSession.cs
using Softing.OPCToolbox.Client;

namespace DABrowse
{
	public class MyDaSession : DaSession
	{

		#region Constructor
		//-----------------

		public MyDaSession(string url) : base(url)
		{
			StateChangeCompleted += new StateChangeEventHandler(HandleStateChanged);
		}

		//--
		#endregion

		#region Private Members
		//---------------------
		bool m_connected = false;

		#region Private Attributes
		//------------------------


		//--
		#endregion

		//--
		#endregion

		#region Public Methods
		//---------------------

		//--
		#endregion

		#region Public Properties
		//-----------------------
		public bool isConnected()
		{
			return m_connected;
		}

		//--
		#endregion

		#region Handles
		//---------------------
		public void HandleStateChanged(ObjectSpaceElement sender, EnumObjectState state)
		{
			if (state == EnumObjectState.CONNECTED)
			{
				m_connected = true;
			}
			else
			{
				m_connected = false;
			}
		}


		//--
		#endregion

	}   //	end class MyDaSession

}   //	end namespace
